src/alert/__tests__/runtime-content.test.tsx

```diff
@@ -0,0 +1,305 @@
+// SPDX-License-Identifier: Apache-2.0
+import * as React from 'react';
+import { act, render } from '@testing-library/react';
+
+import Alert from '../../../lib/components/alert';
+import Button from '../../../lib/components/button';
+import awsuiPlugins from '../../../lib/components/internal/plugins';
+import { awsuiPluginsInternal } from '../../../lib/components/internal/plugins/api';
+import { AlertFlashContentConfig } from '../../../lib/components/internal/plugins/controllers/alert-flash-content';
+import { AlertWrapper } from '../../../lib/components/test-utils/dom';
+
+import stylesCss from '../../../lib/components/alert/styles.css.js';
+
+const pause = (timeout: number) => new Promise(resolve => setTimeout(resolve, timeout));
+
+const expectAlertContent = (
+  wrapper: AlertWrapper,
+  {
+    header,
+    headerReplaced,
+    content,
+    contentReplaced,
+  }: {
+    header?: string | false;
+    headerReplaced?: boolean;
+    content?: string | false;
+    contentReplaced?: boolean;
+  }
+) => {
+  if (header) {
+    if (headerReplaced) {
+      if (wrapper.findHeader()) {
+        expect(wrapper.findHeader()?.getElement()).toHaveClass(stylesCss.hidden);
+      }
+      expect(wrapper.findReplacementHeader()?.getElement().textContent).toBe(header);
+    } else {
+      expect(wrapper.findReplacementHeader()?.getElement()).toHaveClass(stylesCss.hidden);
+      expect(wrapper.findHeader()?.getElement().textContent).toBe(header);
+    }
+  } else if (header === false) {
+    if (wrapper.findHeader()) {
+      expect(wrapper.findHeader()?.getElement()).toHaveClass(stylesCss.hidden);
+    }
+    expect(wrapper.findReplacementHeader()?.getElement()).toHaveClass(stylesCss.hidden);
+  }
+  if (content) {
+    if (contentReplaced) {
+      expect(wrapper.findContent().getElement()).toHaveClass(stylesCss.hidden);
+      expect(wrapper.findReplacementContent().getElement().textContent).toBe(content);
+    } else {
+      expect(wrapper.findReplacementContent().getElement()).toHaveClass(stylesCss.hidden);
+      expect(wrapper.findContent().getElement().textContent).toBe(content);
+    }
+  } else if (content === false) {
+    expect(wrapper.findContent().getElement()).toHaveClass(stylesCss.hidden);
+    expect(wrapper.findReplacementContent().getElement()).toHaveClass(stylesCss.hidden);
+  }
+};
+
+const defaultContent: AlertFlashContentConfig = {
+  id: 'test-content',
+  runReplacer(context, registerReplacement) {
+    registerReplacement('header', container => {
+      container.append('New header');
+    });
+    registerReplacement('content', container => {
+      container.append('New content');
+    });
+    return {
+      update: () => {},
+      unmount: () => {},
+    };
+  },
+};
+
+function delay(advanceBy = 1) {
+  const promise = act(() => new Promise(resolve => setTimeout(resolve)));
+  jest.advanceTimersByTime(advanceBy);
+  return promise;
+}
+
+beforeEach(() => {
+  jest.useFakeTimers();
```

Comment:
Would it be sensible to use waitFor helper instead of fake timers? Like:

```
test('renders runtime content initially', async () => {
  awsuiPlugins.alertContent.registerContentReplacer(defaultContent);
  const { container } = render(<Alert>Alert content</Alert>);
  const alertWrapper = new AlertWrapper(container);
  await waitFor(() => {
    expectAlertContent(alertWrapper, { content: 'New content', contentReplaced: true });
  });
});
```